File: backend/src/main/java/com/levin/backend/cleaning_roster/Room.java
package com.levin.backend.cleaning_roster;

import com.levin.backend.cleaning_roster.model.ImageProperties;
import org.springframework.data.annotation.Id;

import java.util.List;

public record Room(
        @Id
        String id,
        String name,
        ImageProperties imageProperties,
        int rowSpan,
        int columnSpan,
        List<String> assignments
) {

        public Room withId(String id) {
                return new Room(id,
                        this.name,
                        this.imageProperties,
                        this.rowSpan,
                        this.columnSpan,
                        this.assignments
                );
        }

        public Room withAssignments(List<String> assignments) {
                return new Room(
                        this.id,
                        this.name,
                        this.imageProperties,
                        this.rowSpan,
                        this.columnSpan,
                        assignments
                );
        }
}

File: frontend/src/model/Room.ts
export type Room = {
    id: string,
    name: string,
    imageProperties: ImageProperties
    rowSpan: number,
    columnSpan: number,
    assignments: string[]
}

export type ImageProperties = {
    index: number,
    hue: number,
    brightness: number,
    isInverted: boolean
    isBlackAndWhite: boolean,
    isOverSaturated: boolean
}

export const dummyImageProperties: ImageProperties = {
    index: 7,
    hue: 0,
    brightness: 0,
    isInverted: false,
    isBlackAndWhite: false,
    isOverSaturated: false
}

export const dummyRoom: Room = {
    id: "",
    name: "",
    imageProperties: dummyImageProperties,
    rowSpan: 1,
    columnSpan: 1,
    assignments: []
}